spec/function.dd

```diff
@@ -173,124 +173,315 @@ int foo();
 
 $(H2 $(LNAME2 pure-functions, Pure Functions))
 
-        $(P Pure functions are functions that cannot directly access global or static
-            mutable state. `pure` guarantees that a pure function call
-            won't access or modify any implicit state in the program.
-        )
+$(P The `pure` attribute can be specified with the declaration of functions and methods. The
+definition of a pure function is statically checked for additional restrictions as specified
+in this section.)
 
-        $(P Unlike other functional programming languages, D's `pure`
-            functions allow modification of the caller state through their mutable
-            parameters.
-        )
+$(P A function annotated with `pure` cannot contain reads or writes of global or static mutable
+state. Example:)
```

Comment:
what about `shared` state? is it fair to say a pure function can't accept shared arguments?

Reply:
@TurkeyMan I assume `shared` is also mutable. But that reminds me `const` is also disallowed.

Reply:
OK, I changed to wording to only allow `immutable` and `enum` reads.